kubespawner/spawner.py

```diff
@@ -472,7 +511,7 @@ def poll(self):
 
     @gen.coroutine
     def start(self):
-        if self.storage_class:
+        if self.user_storage_class && self.user_storage_capacity:
```

Comment:
'and' not && :) Should also explicitly use a 'is not None' check instead. And an or instead of and probably.